chore: Add checks for template names for build command in cli

JayaKrishnaNamburu opened the pull request:
## Description

Small changes that adds checks for templates and hints for templates in the `--help` command.

## Steps for reproduction

```sh
webstudio-cli --help
```

## Code Review

- [ ] hi @TrySound , I need you to do
  - detailed review (read every line)

TrySound commented on packages/cli/src/commands/build.ts:
```diff
@@ -21,7 +21,9 @@ export const buildOptions = (yargs: CommonYargsArgv) =>
-      describe: "[Experimental] Template to use for the build",
+      describe: `[Experimental] Template to use for the build
+Templates Available
+${PROJECT_TEMPALTES.join(`\n`)}`,
```
You can try `choices: ['xs', 's', 'm', 'l', 'xl']` field

JayaKrishnaNamburu replied:
We can't use the `choices` because local templates will fail the check. As `yargs` itself is throwing error if users pass
```sh
webstudio-cli build --template ./template
```

something like this.

TrySound commented on packages/cli/src/prebuild.ts:
```diff
@@ -198,6 +198,16 @@ export const prebuild = async (options: {
 }) => {
+  if (
+    options.template !== undefined &&
+    options.template.startsWith("./") === false &&
```
Just `.`. User may refer to parent folder.

kof commented on packages/cli/src/commands/init-flow.ts:
```diff
@@ -119,3 +112,7 @@ export const initFlow = async (
   );
 };
+
+export const convertToTitleCase = (str: string) => {
```
we are using in builder import { titleCase } from "title-case";

kof replied:
and we also have a humanizeString utility, but I guess you only need title-case

JayaKrishnaNamburu replied:
But the package seems to change the `netlify-functions` to `
Netlify-Functions` but we are displaying like `Netlify Functions`. Is it fine ? I don't have much of a opinion on how to show. But the package claims to use english rules. So, sounds promising, changing it now 👍

JayaKrishnaNamburu commented:
Renamed `netlify-edge-functions` to `netlify`. So people can use edge as default. And the same reflects in `init-flow` too. The options will be

```sh
[ ] Vercel
[ ] Netlify
[ ] Netlify Functions
```

TrySound commented on packages/cli/src/config.ts:
```diff
@@ -51,3 +51,5 @@ export const jsonToGlobalConfig = (json: unknown) => {
 
 export type GlobalConfig = z.infer<typeof zGlobalConfig>;
+
+export const PROJECT_TEMPALTES = ["vercel", "netlify", "netlify-functions"];
```
Were both functions? The distinction was in "edge", no?

JayaKrishnaNamburu replied:
Yes, and i think the decisions was `edge` is cheaper and efficient make it default. We can add the differences between these in docs.

TrySound replied:
I mean netlify-functions does not describe at all. Both netlify options are functions

JayaKrishnaNamburu replied:
Switched back to `netlify-functions` and `netlify-edge-functions` from their platform

istarkov commented on .github/workflows/cli-test.yml:
```diff
@@ -69,17 +69,17 @@ jobs:
         # Testing fixtures for netlify-edge-functions template
-      - name: Test cli link command for netlify-edge-functions
-        working-directory: ./fixtures/webstudio-remix-netlify-edge-functions
+      - name: Test cli link command for netlify
```
1st lets move this into fixture  packages
i.e.
```json
"scripts": {
    "fixture:link": "webstudio-cli link --link 'https://webstudio-builder-git-main-webstudio-is.vercel.app/builder/d845c167-ea07-4875-b08d-83e97c09dcce?authToken=e9d1343f-9298-4fd3-a66e-f89a5af2dd93",
    "fixture:sync": "blabla",
    "fixture:build": "blablablabla",
...
}
    ```

then using pnpm just run from the root 
```bash
pnpm --filter='./fixtures/*' run --parallel fixtures:link
pnpm --filter='./fixtures/*' run --parallel fixtures:sync
pnpm --filter='./fixtures/*' run --parallel fixtures:build
```

to never touch test.yaml again

istarkov reviewed (commented):
plz do so we would able to resync rebuild relink all fixtures from the root

istarkov commented on packages/cli/src/config.ts:
```diff
@@ -51,3 +51,5 @@ export const jsonToGlobalConfig = (json: unknown) => {
 
 export type GlobalConfig = z.infer<typeof zGlobalConfig>;
+
+export const PROJECT_TEMPALTES = ["vercel", "netlify", "netlify-functions"];
```
Isn't this just a folder names, use them

istarkov replied:
Just any time we will add a new template to not search why its not shown

istarkov reviewed (approved):
plz do

JayaKrishnaNamburu commented:
`custom-template` fixture is broken. Not it's corrected. See the `projectId` inside the `cli-test.yml` file is not matching with the projectId inside the fixture. Corrected it and updated the fixture too. 
![Screenshot 2023-09-25 at 19 48 34](https://github.com/webstudio-is/webstudio/assets/11075561/55ed05c7-58cb-420c-8345-a76e8a0be7f4)
![Screenshot 2023-09-25 at 19 48 52](https://github.com/webstudio-is/webstudio/assets/11075561/0acaa690-9aba-4b1a-8c15-c5e6b05803fd)

https://github.com/webstudio-is/webstudio/pull/2358/files#diff-69532840426f93d7d6b29deb0ff774e32affeb262dc7c7145e1f395d5c69513a

https://github.com/webstudio-is/webstudio/pull/2358/files#diff-5fcc33269241517d79034d41d1a2f0ce649c35812ea21326ee55a45ee3413f0cL87
